cupy/cuda/compiler.py

```diff
@@ -25,18 +25,25 @@ class NVCCException(Exception):
     pass
 
 
-def _run_nvcc(cmd, cwd):
+def _run_nvcc(cmd, cwd, log_stream=None):
     try:
-        return subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
+        stdout = subprocess.check_output(cmd, cwd=cwd,
+                                         stderr=subprocess.STDOUT,
+                                         universal_newlines=True)
+        if log_stream == 'stdout':
+            print(stdout)
+        elif log_stream is not None:
+            with open(log_stream, 'w') as f:
+                f.write(stdout)
+        return stdout
     except subprocess.CalledProcessError as e:
         msg = ('`nvcc` command returns non-zero exit status. \n'
                'command: {0}\n'
                'return-code: {1}\n'
                'stdout/stderr: \n'
                '{2}'.format(e.cmd,
                             e.returncode,
-                            e.output.decode(encoding='UTF-8',
-                                            errors='replace')))
+                            e.output))
```

Comment:
Could you comment why we need this change?

Reply:
Ahh, yes. By adding `universal_newlines=True` to `check_output` we are creating a string. Therefore there is no need to `decode`. 

Here is the error without the code change
```bash
                   '{2}'.format(e.cmd,
                                e.returncode,
>                               e.output.decode(encoding='UTF-8',
                                                errors='replace')))
E           AttributeError: 'str' object has no attribute 'decode'
```

Reply:
Ah, good catch! Thanks Matt.